src/core/server/plugins/plugin_context.ts

```diff
@@ -65,6 +65,13 @@ export function createPluginInitializerContext(
      * Core configuration functionality, enables fetching a subset of the config.
      */
     config: {
+      /**
+       * Global configuration
+       * Note: naming not final here, it will be renamed in a near future (https://github.com/elastic/kibana/issues/46240)
+       * @deprecated
+       */
+      globalConfig__deprecated$: coreContext.configService.getConfig$(),
```

Comment:
According to @joshdover's comment in the original issue,
> make it obvious in the naming that this will go away
that's why I went for that ugly and aggressive name

Reply:
I just remembered that we already have similar situation in `elasticsearch` service. Would be good to use the same pattern
https://github.com/elastic/kibana/blob/master/src/core/server/elasticsearch/elasticsearch_service.ts#L98